src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java

```diff
@@ -67,31 +115,46 @@ void testLogsMappersIsCalled() throws Exception {
-            Mockito.when(networkStoreServiceMocked.getNetwork(eq(networkUuid), any(PreloadingStrategy.class))).thenReturn(networkMocked);
-            Mockito.when(networkMocked.getVariantManager()).thenReturn(variantManagerMocked);
-            Mockito.when(networkMocked.getBusView()).thenReturn(busViewMocked);
-            Mockito.when(busViewMocked.getBusStream()).thenAnswer(invocation -> Stream.empty());
-            Mockito.when(reportMapperMocked.processReporter(any(Reporter.class))).thenReturn(reporter);
-            final ShortCircuitWorkerService workerService = new ShortCircuitWorkerService(networkStoreServiceMocked, reportServiceMocked, shortCircuitExecutionService, notificationServiceMocked, resultRepositoryMocked, objectMapperMocked, List.of(reportMapperMocked), shortCircuitObserver);
+            shortCircuitResultContextMockedStatic.when(() -> ShortCircuitResultContext.fromMessage(message, objectMapper)).thenReturn(resultContext);
+            when(networkStoreService.getNetwork(eq(networkUuid), any(PreloadingStrategy.class))).thenReturn(networkMocked);
+            when(networkMocked.getVariantManager()).thenReturn(variantManagerMocked);
+            when(networkMocked.getBusView()).thenReturn(busViewMocked);
+            when(busViewMocked.getBusStream()).thenAnswer(invocation -> Stream.empty());
+            when(reportMapper.processReporter(any(Reporter.class))).thenReturn(reporter);
             workerService.consumeRun().accept(message);
             shortCircuitAnalysisMockedStatic.verify(ShortCircuitAnalysis::find, atLeastOnce());
-            Mockito.verify(reportMapperMocked, times(1)).processReporter(any(ReporterModel.class));
-            Mockito.verify(reportServiceMocked, times(1)).sendReport(reportUuid, reporter);
+            verify(reportMapper, times(1)).processReporter(any(ReporterModel.class));
+            verify(reportService, times(1)).sendReport(reportUuid, reporter);
         }
     }
 
-    @AllArgsConstructor
-    private static class ShortCircuitAnalysisProviderMock implements ShortCircuitAnalysisProvider {
-        private final ShortCircuitAnalysisResult result;
+    @Test
+    void testGetBusFaultFromOutOfVoltageBus() {
+        var network = mock(Network.class);
+        var context = mock(ShortCircuitRunContext.class);
+        var busbarSection = mock(BusbarSection.class);
+        var terminal = mock(Terminal.class);
+        var busView = mock(BusView.class);
+        var busId = "bus1";
+
+        when(context.getBusId()).thenReturn(busId);
+        doReturn(busbarSection).when(network).getIdentifiable(busId);
+        when(busbarSection.getTerminal()).thenReturn(terminal);
+        when(terminal.getBusView()).thenReturn(busView);
+        when(busView.getBus()).thenReturn(null);
+
+        var exception = (ShortCircuitException) catchRuntimeException(() -> workerService.getBusFaultFromBusId(network, context));
+        assertThat(exception.getType()).isEqualTo(BUS_OUT_OF_VOLTAGE);
```

Comment:
@etiennehomer 
I'm not sure we should use "var", what do you think ?
In such a small scope, I guess it's ok

Reply:
Yes why not in a small test method like this